src/TraceEvent/TraceEventSession.cs

```diff
@@ -1116,6 +1116,22 @@ public void CaptureState(Guid providerGuid, ulong matchAnyKeywords = ulong.MaxVa
                     asArray[6] = (byte)(longVal >> 48);
                     asArray[7] = (byte)(longVal >> 56);
                 }
+                // Query existing keywords and merge them with requested keywords before capture state
+                ulong mergedKeywords = matchAnyKeywords;
+                Guid* providerPtr = stackalloc Guid[1];
```

Comment:
Instead of using `stackalloc`, you should be able to use the fixed keyword to convert the `Guid` to a `Guid*`.